src/OpenTelemetry/Metrics/MeterSdk.cs

```diff
@@ -159,8 +159,18 @@ public virtual void Collect()
                         var aggregator = handle.Value.GetAggregator();
                         aggregator.Checkpoint();
                         var metricData = aggregator.ToMetricData();
-                        metricData.Labels = labelSet.Labels;
-                        metric.Data.Add(metricData);
+
+                        var haveData = true;
+                        if (metricData is Int64SummaryData summary)
+                        {
+                            haveData = summary.Count > 0;
+                        }
+
+                        if (haveData)
+                        {
+                            metricData.Labels = labelSet.Labels;
+                            metric.Data.Add(metricData);
+                        }
```

Comment:
Do we have C# 9 already in?

```suggestion
                        if (metricData is Int64SummaryData {Count>0} summary)
                        {
                            metricData.Labels = labelSet.Labels;
                            metric.Data.Add(metricData);
                        }
```

Reply:
I assume we can use C# 9 syntax.  And if not, our CI should fail.

In this case, I was being extra careful in the case that metricData is NOT an Int64SummaryData.  Thus, I am "filtering" only if I am positively assured it's a SummaryData with count=0.  Otherwise, I don't want to alter behavior.

I think we are generally not handling "empty" collect/export cycles correctly.  We still emit "empty" records (for both Measures and Sum).  Ideally, I think we would not even create a new Metric (with Data.Count = 0) to pass to Processors.  In the SumData case, I think we need to include a Count field, so we can distinguish between having/not-having reported data.  (i.e. Empty case (sum=0, 0 data point) vs multiple data points where sum=0)

However, I am new to this code base, so please provide feedback for such.

Reply:
oh I see what you did (and I broke) now. I haven't been reviewing code here (or writing) in months so, better wait for someone else to give proper feedback here.

Reply:
another option:
```csharp
 if (metricData is Int64SummaryData summary && summary.Count > 0)
                        {
                            metricData.Labels = labelSet.Labels;
                            metric.Data.Add(metricData);
                        }
```

Reply:
The catch here is that he started with `var haveData = true;`. So he wants to add the value by default, but in case it's `Int64SummaryData`, only if it's `Count>0`.

Reply:
Would it be helpful to add `HasData` on the `MetricData` base class?
```csharp
public abstract class MetricData
{
   public virtual bool HasData => true;
}

public class Int64SummaryData
{
   public override bool HasData => Count > 0;
}
```

Allowing for this code here to just be:
```csharp
                        if (metricData.HasData)
                        {
                            metricData.Labels = labelSet.Labels;
                            metric.Data.Add(metricData);
                        }
```

Reply:
Yes, I would also agree.  We currently cannot distinguish an "empty" Int64SummaryData (and also Int64SumData).  If we added HasData, it will definitely be easier.  This will also help with us emitting unnecessary "empty" Int64SumData as well.  I will refactor accordingly.